src/components/AutocompleteDropdown.tsx

```diff
@@ -4,12 +4,13 @@ import clsx from 'clsx'
 import useBooleanState from '../hooks/useBooleanState'
 import { Sentry } from '@oneblink/apps'
 
-type AutocompleteOption = {
+type AutocompleteOption<T> = {
   label: string
   value: string
+  data: T | undefined
```

Comment:
if you make this 

```suggestion
  data?: T
```

There should be no changes required to anywhere this component is currently being used.